core/src/Streamly/Internal/Data/Stream/StreamK.hs

```diff
@@ -1380,6 +1406,115 @@ parseChunks :: (Monad m, Unbox a) =>
+            ParserK.Continue n cont1 -> do
+                let n1 = negate n
+                assertM(n1 >= 0 && n1 <= length backBuf)
+                let (s1, backBuf1) = backTrack n1 backBuf nil
+                 in go backBuf1 cont1 s1
+            ParserK.Done 0 b ->
+                return (Right b, nil)
+            ParserK.Done n b -> do
+                let n1 = negate n
+                assertM(n1 >= 0 && n1 <= length backBuf)
+                let (s1, _) = backTrack n1 backBuf nil
+                 in return (Right b, s1)
+            ParserK.Error err -> return (Left (ParseError err), nil)
+
+    seekErr n =
+        error $ "parseBreak: Partial: forward seek not implemented n = "
+            ++ show n
+
+    yieldk
+        :: [a]
+        -> (ParserK.Input a -> m (ParserK.Step a m b))
+        -> a
+        -> StreamK m a
+        -> m (Either ParseError b, StreamK m a)
+    yieldk backBuf parserk arr stream = do
+        pRes <- parserk (ParserK.Single arr)
+        case pRes of
+            ParserK.Partial 1 cont1 -> go [] cont1 stream
+            ParserK.Partial 0 cont1 -> go [] cont1 (cons arr stream)
+            ParserK.Partial n _ | n > 1 -> seekErr n
+            ParserK.Partial n cont1 -> do
+                let n1 = negate n
+                    bufLen = length backBuf
+                    s = cons arr stream
+                assertM(n1 >= 0 && n1 <= bufLen)
+                let (s1, _) = backTrack n1 backBuf s
+                go [] cont1 s1
+            ParserK.Continue 1 cont1 -> go (arr:backBuf) cont1 stream
+            ParserK.Continue 0 cont1 ->
+                go backBuf cont1 (cons arr stream)
+            ParserK.Continue n _ | n > 1 -> seekErr n
+            ParserK.Continue n cont1 -> do
+                let n1 = negate n
+                    bufLen = length backBuf
+                    s = cons arr stream
+                assertM(n1 >= 0 && n1 <= bufLen)
+                let (s1, backBuf1) = backTrack n1 backBuf s
+                go backBuf1 cont1 s1
+            ParserK.Done 1 b -> pure (Right b, stream)
+            ParserK.Done 0 b -> pure (Right b, cons arr stream)
+            ParserK.Done n _ | n > 1 -> seekErr n
+            ParserK.Done n b -> do
+                let n1 = negate n
+                    bufLen = length backBuf
+                    s = cons arr stream
+                assertM(n1 >= 0 && n1 <= bufLen)
+                let (s1, _) = backTrack n1 backBuf s
+                pure (Right b, s1)
+            ParserK.Error err -> return (Left (ParseError err), nil)
+
+    go
+        :: [a]
+        -> (ParserK.Input a -> m (ParserK.Step a m b))
+        -> StreamK m a
+        -> m (Either ParseError b, StreamK m a)
+    go backBuf parserk stream = do
+        let stop = goStop backBuf parserk
+            single a = yieldk backBuf parserk a nil
+         in foldStream
+                defState (yieldk backBuf parserk) single stop stream
+
+{-# INLINE parse #-}
+parse :: Monad m =>
```

Comment:
Document this.

Please use `parseChunks` where possible, for better performance.